Intermittent simulator crash from UnicodeDecodeError

I am seeing an intermittent error at simulator start up when using `xcellium`

```
xrun(64): 18.03-s006: (c) Copyright 1995-2018 Cadence Design Systems, Inc.
Loading snapshot worklib.aes_gcm_top:sv .................... Done
xmsim: *W,DSEM2009: This SystemVerilog design is simulated as per IEEE 1800-2009 SystemVerilog simulation semantics. Use -disable_sem2009 option for turning off SV 2009 simulation semantics.
     -.--ns INFO     cocotb.gpi                                  gpi_embed.c:78   in set_program_name_in_venv        Did not detect Python virtual environment. Using system-wide Python interpreter
     -.--ns INFO     cocotb.gpi                                GpiCommon.cpp:104  in gpi_print_registered_impl       VPI registered
xcelium> source /cad/adi/apps/cadence/xlm/linux/18.03-s06/tools.lnx86/inca/files/xmsimrc
xcelium> run
UnicodeDecodeError: 'utf-8' codec can't decode bytes in position 0-1: invalid continuation byte
UnicodeDecodeError: 'utf-8' codec can't decode bytes in position 47-48: invalid continuation byte
     0.00ns ERROR    cocotb.gpi                                gpi_logging.c:224  in gpi_log                         Error calling Python logging function from C
```

Attaching the simulator log if it is any help.

This is the error log from [gh-1364](https://github.com/cocotb/cocotb/pull/1364)

```
xmsim: *W,DSEM2009: This SystemVerilog design is simulated as per IEEE 1800-2009 SystemVerilog simulation semantics. Use -disable_sem2009 option for turning off SV 2009 simulation semantics.
     -.--ns INFO     cocotb.gpi                                  gpi_embed.c:79   in set_program_name_in_venv        Did not detect Python virtual environment. Using system-wide Python interpreter
     -.--ns INFO     cocotb.gpi                                GpiCommon.cpp:104  in gpi_print_registered_impl       VPI registered
xcelium> source /cad/adi/apps/cadence/xlm/linux/18.03-s06/tools.lnx86/inca/files/xmsimrc
xcelium> run
     -.--ns WARNING  cocotb.gpi                                  gpi_embed.c:342  in embed_sim_init                  Unable to decode argv[4] = b'0\xe9\x84\x04' as utf8, leaving undecoded as bytes
     -.--ns ERROR    cocotb.gpi                                gpi_logging.c:244  in gpi_log_v                       Error calling Python logging function from C while logging the above
UnicodeDecodeError: 'utf-8' codec can't decode bytes in position 1-2: invalid continuation byte

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/tmp/cocotb_eric/cocotb/log.py", line 178, in _filter_from_c
    return logging.getLogger(logger_name).isEnabledFor(level)
  File "/tmp/anaconda3/envs/cocotb_dev/lib/python3.7/logging/__init__.py", line 1932, in getLogger
    return Logger.manager.getLogger(name)
  File "/tmp/anaconda3/envs/cocotb_dev/lib/python3.7/logging/__init__.py", line 1228, in getLogger
    if not isinstance(name, str):
SystemError: <built-in function isinstance> returned a result with an error set
     0.00ns INFO     cocotb.gpi                                  gpi_embed.c:372  in embed_sim_init                  Running on xmsim(64) version 18.03-s006
     0.00ns INFO     cocotb.gpi                                  gpi_embed.c:373  in embed_sim_init                  Python interpreter initialized and cocotb loaded!
     0.00ns INFO     cocotb                                      __init__.py:138  in _initialise_testbench           Running tests with cocotb v1.4.0.dev0 from /tmp/cocotb_eric
Traceback (most recent call last):
  File "/tmp/cocotb_eric/cocotb/__init__.py", line 142, in _initialise_testbench
    process_plusargs()
  File "/tmp/cocotb_eric/cocotb/__init__.py", line 213, in process_plusargs
    if option.startswith('+'):
TypeError: startswith first arg must be bytes or a tuple of bytes, not str
     0.00ns ERROR    cocotb.gpi                                  gpi_embed.c:457  in embed_sim_init                  cocotb initialization failed - exiting
     0.00ns ERROR    cocotb.scheduler                            __init__.py:198  in _sim_event                      Failing test at simulator request before test run completion: Simulator shutdown prematurely
# check that the file was actually created, since we can't set an exit code from cocotb
test -f results.xml
make[1]: *** [results.xml] Error 1
```

--- a/cocotb/share/lib/embed/gpi_embed.c
+++ b/cocotb/share/lib/embed/gpi_embed.c
@@ -307,7 +307,7 @@ int embed_sim_init(gpi_sim_info_t *info)
         PyObject *argv_item = PyUnicode_DecodeLocale(info->argv[i], "surrogateescape");  // New reference
         if (argv_item == NULL) {
             PyErr_Print();
-            LOG_ERROR("Unable to decode argv item");
+            LOG_ERROR("Unable to convert command line argument %d to Unicode string.", i);
             Py_DECREF(argv_list);
             goto cleanup;
         }

--- a/cocotb/share/lib/embed/gpi_embed.c
+++ b/cocotb/share/lib/embed/gpi_embed.c
@@ -302,10 +302,20 @@ int embed_sim_init(gpi_sim_info_t *info)
         goto cleanup;
     }
     for (i = 0; i < info->argc; i++) {
-        PyObject *argv_item = PyUnicode_FromString(info->argv[i]);      // New reference
+        PyObject *argv_item_raw = PyBytes_FromString(info->argv[i]);       // New reference
+        if (argv_item_raw == NULL) {
+            PyErr_Print();
+            LOG_ERROR("Unable to create argv item");
+            Py_DECREF(argv_list);
+            goto cleanup;
+        }
+        // Decode, embedding non-decodable bytes using PEP-383. This can only
+        // fail with MemoryError or similar.
+        PyObject *argv_item = PyUnicode_FromEncodedObject(argv_item_raw, NULL, "surrogateescape");
+        Py_DECREF(argv_item_raw);
         if (argv_item == NULL) {
             PyErr_Print();
-            LOG_ERROR("Unable to create Python string from argv[%d] = \"%s\"", i, info->argv[i]);
+            LOG_ERROR("Unable to decode argv item");
             Py_DECREF(argv_list);
             goto cleanup;
         }

--- a/cocotb/share/lib/embed/gpi_embed.c
+++ b/cocotb/share/lib/embed/gpi_embed.c
@@ -302,17 +302,9 @@ int embed_sim_init(gpi_sim_info_t *info)
         goto cleanup;
     }
     for (i = 0; i < info->argc; i++) {
-        PyObject *argv_item_raw = PyBytes_FromString(info->argv[i]);       // New reference
-        if (argv_item_raw == NULL) {
-            PyErr_Print();
-            LOG_ERROR("Unable to create argv item");
-            Py_DECREF(argv_list);
-            goto cleanup;
-        }
         // Decode, embedding non-decodable bytes using PEP-383. This can only
         // fail with MemoryError or similar.
-        PyObject *argv_item = PyUnicode_FromEncodedObject(argv_item_raw, NULL, "surrogateescape");
-        Py_DECREF(argv_item_raw);
+        PyObject *argv_item = PyUnicode_DecodeLocale(info->argv[i], "surrogateescape");  // New reference
         if (argv_item == NULL) {
             PyErr_Print();
             LOG_ERROR("Unable to decode argv item");

